app/src/main/java/net/sourceforge/opencamera/cameracontroller/CameraController2.java

```diff
@@ -1244,11 +1252,46 @@ else if( red_i > blue_i ) {
         return temperature;
     }
 
+    private class OnVideoFrameImageAvailableListener implements ImageReader.OnImageAvailableListener {
+        @Override
+        public void onImageAvailable(ImageReader reader) {
+            if (MyDebug.LOG) {
+                Log.d(TAG, "Frame during video recording");
+            }
+            Image image = reader.acquireNextImage();
+
+            if (MyDebug.LOG)
+                Log.d(TAG, "video frame timestamp: " + image.getTimestamp());
+            if (MyDebug.LOG)
+                Log.d(TAG, "image format: " + image.getFormat());
+
+            long timestamp = image.getTimestamp();
+            if (mWantSaveFrames) {
+                byte[] nv21 = YuvImageUtils.YUV420toNV21(image);
```

Comment:
Not obvious: any reasons to convert to intermediate nv21 if we really need JPEG later?

Reply:
We need it because YUV420_888 can represent [multiple color formats](https://developer.android.com/reference/android/media/MediaCodecInfo.CodecCapabilities#COLOR_FormatYUV420Flexible), but to get jpeg or bitmap we need to convert Image planes to this particular byte representation in NV21 format